### PR TITLE
Add support for switching cipher methods

### PR DESCRIPTION
**Pull request by seba-aln**
feat: Option to change the ciphering method

Introduced option to select ciphering method for encoding messages and files. The default behavior is unchanged. More can be read [in this comment](https://github.com/pubnub/python/pull/156#issuecomment-1623307799)

### CODE REVIEW

File: examples/crypto.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+from Cryptodome.Cipher import AES
```
Comment (marek-lewandowski) [inline examples/crypto.py:1]: ;) 
https://peps.python.org/pep-0008/#imports

**Review by kleewho — APPROVED** [review]
LGTM

**Comment by seba-aln** [comment]
## This is potentially a breaking change 
If you use multiple clients, enabling AES.GCM will break other clients that have not enabled this because they will not be able to decipher the messages.

If the python SDK is the only one in use, there should not be any problems since history messages can be deciphered by a fallback algorithm.

## Configuration
Nothing has to be changed for this to work with other SDKs without breaking compatibility. If, however, you wish to enable GCM ciphering, use the example below:

```
from Cryptodome.Cipher import AES
...
config = PNConfiguration()
config.cipher_mode = AES.MODE_GCM
config.fallback_cipher_mode = AES.MODE_CBC
```

**Comment by seba-aln** [comment]
@pubnub-release-bot release

**Pull request merged**